src/context.js

```diff
@@ -14,8 +15,13 @@ type ProviderState = {
 class LoadsProvider extends React.Component<ProviderProps, ProviderState> {
   state = { data: {} };
 
-  setResponse = (key: string, { response = null, error = null }: { response?: any, error?: any }) => {
-    this.setState({ data: { ...this.state.data, [key]: { response, error } } });
+  setResponse = (key: string, { response, error, state }: { response?: any, error?: any, state: string }) => {
```

Comment:
You refactored another function to expect enum type for state, do the same here?

Reply:
Sounds good